Python/pywarpx/diagnostics/data/ParticleExtrema.py

```diff
@@ -0,0 +1,69 @@
+
+from BaseReader import DataReader
+
+
+class ParticleExtremaData(DataReader):
+    """
+    Reader for the ParticleExtrema reduced diagnostic.
+    """
+
+    def __init__(self, red_diags_dir="./diags/reducedfiles", separator=" ", fname='ParticleExtrema.txt'):
+        """
+        Parameters
+        ----------
+        red_diags_dir: string
+            Path to the reduced diagnostic directory, default is './diags/reducedfiles'.
+        fname: string
+            Name of the reduced diagnostic file.
+        separator: string
+            The separator between row values in the output file, default is a whitespace " ".
+        """
+
+        super().__init__(red_diags_dir, separator)
+
+        self.fname = fname
+        if self.fname is None:
+            raise ValueError('The diagnostic file name cannot be None!')
+
+        self.has_species = False
+
+    def get_valid_args(self):
+        """
+        Returns the valid strings to extract the data.
+        """
+
+        if self.get_ncols() == 18:
+            valid_args = ('xmin', 'xmax', 'ymin', 'ymax', 'zmin', 'zmax', 'pxmin', 'pxmax', 'pymin', 'pymax', 'pzmin', 'pzmax', 'gmin', 'gmax', 'wmin', 'wmax')
+        elif self.get_ncols() == 20:
+            valid_args = ('xmin', 'xmax', 'ymin', 'ymax', 'zmin', 'zmax', 'pxmin', 'pxmax', 'pymin', 'pymax', 'pzmin', 'pzmax', 'gmin', 'gmax', 'wmin', 'wmax' 'chimin', 'chimax')
```

Comment:
You can add an `else` block that raises an error if the number of columns isn't what was expected.